ignite/handlers/state_param_scheduler.py

```diff
@@ -0,0 +1,390 @@
+
+            # basic handler to print scheduled state parameter
+            engine.add_event_handler(Events.EPOCH_COMPLETED, lambda _ : print(engine.state.param))
+
+            engine.run([0] * 8, max_epochs=2)
+
+    .. versionadded:: 0.6.0
+
+    """
+
+    def __init__(self, lambda_obj: Any, param_name: str, save_history: bool = False):
+        super(LambdaStateScheduler, self).__init__(param_name, save_history)
+
+        if not callable(lambda_obj):
+            raise ValueError("Expected lambda_obj to be callable.")
+
+        self.lambda_obj = lambda_obj
+        self._state_attrs += ["lambda_obj"]
+
+    def get_param(self) -> Union[List[float], float]:
+        return self.lambda_obj(self.event_index)
+
+
+class PiecewiseLinearStateScheduler(StateParamScheduler):
+    """Piecewise linear state parameter scheduler.
+
+    Args:
+        milestones_values: list of tuples (event index, parameter value)
+            represents milestones and parameter values. Milestones should be increasing integers.
+        param_name: name of parameter to update.
+        save_history: whether to log the parameter values to
+            `engine.state.param_history`, (default=False).
+
+    Examples:
+
+        .. code-block:: python
+
+            ...
+            engine = Engine(train_step)
+
+            param_scheduler = PiecewiseLinearStateScheduler(
+                param_name="param",
+                milestones_values=[(10, 0.5), (20, 0.45), (21, 0.3), (30, 0.1), (40, 0.1)]
+            )
+
+            param_scheduler.attach(engine, Events.EPOCH_COMPLETED)
+
+            # basic handler to print scheduled state parameter
+            engine.add_event_handler(Events.EPOCH_COMPLETED, lambda _ : print(engine.state.param))
+
+            engine.run([0] * 8, max_epochs=40)
+            #
+            # Sets the state parameter value to 0.5 over the first 10 iterations, then decreases linearly from 0.5 to
+            # 0.45 between 10th and 20th iterations. Next there is a jump to 0.3 at the 21st iteration and the state
+            # parameter value decreases linearly from 0.3 to 0.1 between 21st and 30th iterations and remains 0.1 until
+            # the end of the iterations.
+            #
+
+    .. versionadded:: 0.6.0
+    """
+
+    def __init__(
+        self, milestones_values: List[Tuple[int, float]], param_name: str, save_history: bool = False,
+    ):
+        super(PiecewiseLinearStateScheduler, self).__init__(param_name, save_history)
+
+        values = []  # type: List[float]
+        milestones = []  # type: List[int]
+        for pair in milestones_values:
+            if not isinstance(pair, tuple) or len(pair) != 2:
+                raise ValueError("Argument milestones_values should be a list of pairs (milestone, param_value)")
+            if not isinstance(pair[0], numbers.Integral):
+                raise TypeError(f"Value of a milestone should be integer, but given {type(pair[0])}")
```

Comment:
@fco-dv The exceptions seems still not covered. Could you check please ?

Reply:
added a test `test_piecewiselinear_asserts`